### PR TITLE
RFP/securityMarketPlace

### PR DESCRIPTION
**Pull request by ParthChaudhary31**
#### Project Abstract

> Please replace these instructions with a brief description of your project summarising key points (1-2 paragraphs).
>
> If your application is a follow-up to a previous grant, please mention which one in the first line of the abstract and include a link to previous pull requests if applicable.

#### Grant [level](https://github.com/w3f/Grants-Program#level_slider-levels)

- [ ] **Level 1**:  Up to $10,000, 2 approvals
- [x] **Level 2**:  Up to $30,000, 3 approvals
- [ ] **Level 3**:  Unlimited, 5 approvals (for >$100k: Web3 Foundation Council approval)

#### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied and aptly renamed (`project_name.md`).
- [x] I have read the [application guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/grant_guidelines_per_category.md).
- [x] Payment details have been provided (bank details via email _or_ BTC, Ethereum (USDC/DAI) or Polkadot/Kusama (USDT) address in the application).
- [x] The software delivered for this grant will be released under an open-source license specified in the application.
- [x] The initial PR contains only one commit (squash and force-push if needed).
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process) (see the [announcement guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/announcement-guidelines.md)).
- [x] I prefer the discussion of this application to take place in a private Element/Matrix channel. My username is: `@_______:matrix.org` (change the homeserver if you use a different one)

### CODE REVIEW

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for your interest in our grants program. Could you update the formatting of the milestone tables? (see our [template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md)). The milestone tables should also specify the technical deliveries in a way that we can evaluate them. The FTE, for example, isn't something that we can double-check. Also are you aware of the following application: https://github.com/w3f/Grants-Program/pull/1590

**Comment by ParthChaudhary31** [comment]
> Thanks for your interest in our grants program. Could you update the formatting of the milestone tables? (see our [template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md)). The milestone tables should also specify the technical deliveries in a way that we can evaluate them. The FTE, for example, isn't something that we can double-check. Also are you aware of the following application: #1590

I made the changes to the table as per requested and yes, We are aware of #1590 's proposal, They have proposed a solution entirely based on smart contracts while we abstracted a lot of functionalities and have proposed to solve them through node services to make the platform optimal for gas fees while using smart contracts in only the most essential parts to keep the platform truly decentralized.

**Comment by ParthChaudhary31** [comment]
> Thanks for your interest in our grants program. Could you update the formatting of the milestone tables? (see our [template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md)). The milestone tables should also specify the technical deliveries in a way that we can evaluate them. The FTE, for example, isn't something that we can double-check. Also are you aware of the following application: #1590

I made the changes to the table as per requested and yes, We are aware of #1590 's proposal, They have proposed a solution entirely based on smart contracts while we abstracted a lot of functionalities and have proposed to solve them through node services to make the platform optimal for gas fees while using smart contracts in only the most essential parts to keep the platform truly decentralized.

**Comment by ParthChaudhary31** [comment]
> Thanks for your interest in our grants program. Could you update the formatting of the milestone tables? (see our [template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md)). The milestone tables should also specify the technical deliveries in a way that we can evaluate them. The FTE, for example, isn't something that we can double-check. Also are you aware of the following application: #1590

I made the changes to the table as per requested and yes, We are aware of #1590 's proposal, They have proposed a solution entirely based on smart contracts while we abstracted a lot of functionalities and have proposed to solve them through node services to make the platform optimal for gas fees while using smart contracts in only the most essential parts to keep the platform truly decentralized.

**Comment by ParthChaudhary31** [comment]
]

**Comment by ParthChaudhary31** [comment]
@semuelle Are there any further pending changes you'd like us to update ?

**Review by semuelle — COMMENTED** [review]
@ParthChaudhary31, to expand on what @Noc2 said: the architecture is something we usually expect the applicant to already have largely thought out, so M1 wouldn't be covered. And since we expect every milestone to come with the default deliverables, including tests, the third milestone doesn't work either. 

I suggest you split M2 into two milestones, each with the default deliverables (tests, documentation, etc.) and remove M1 and M3. It would also be good to see how the smart contracts and backend interact with another, i.e. which data and functionality remains if the servers are shut down.

FYI, in your lifecycle description you speak only of smart contracts. The audits could be of any other project, though.

**Comment by ParthChaudhary31** [comment]
> Thanks for the update. However, all milestone tables should contain the default deliverables 0.a to 0.d (https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md#milestone-1-example--basic-functionality) and your specific deliverables as 1, 2, etc. Also, your first milestone is usually not something that we support via our grants program since it's not useful for others. Finally, what do you mean by "finish the development of the front end and turn it into the interactive Front-End.". The specification of the milestone tables are the requirements of our contracts, so we need to be able to evaluate them. "Finish the front-end" could, for example, mean anything.

@Noc2 Thanks for your suggestions, we have updated the milestones accordingly and tried to make our deliverables more specific.

**Comment by ParthChaudhary31** [comment]
> @ParthChaudhary31, to expand on what @Noc2 said: the architecture is something we usually expect the applicant to already have largely thought out, so M1 wouldn't be covered. And since we expect every milestone to come with the default deliverables, including tests, the third milestone doesn't work either.
> 
> I suggest you split M2 into two milestones, each with the default deliverables (tests, documentation, etc.) and remove M1 and M3. It would also be good to see how the smart contracts and backend interact with another, i.e. which data and functionality remains if the servers are shut down.
> 
> FYI, in your lifecycle description you speak only of smart contracts. The audits could be of any other project, though.

@semuelle Thanks for your suggestions, we have updated the milestones accordingly.

We were also aware that audits can be of any kind and have altered our wording to make it more inclusive.

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the update. The application looks much better now. I have two follow up questions:
- Could you specify the ink smart contracts in more detail as part of the milestone table specification? These are the requirements of our contracts. 
- Could you define "We will deliver the completely integrated and working product." in more detail? Feel free to also reference your previous explanation here. 

Finally one minor thing: You can remove "Example" after Milestones 1 and 2.

**Comment by ParthChaudhary31** [comment]
> Thanks for the update. The application looks much better now. I have two follow up questions:
> 
> * Could you specify the ink smart contracts in more detail as part of the milestone table specification? These are the requirements of our contracts.
> * Could you define "We will deliver the completely integrated and working product." in more detail? Feel free to also reference your previous explanation here.
> 
> Finally one minor thing: You can remove "Example" after Milestones 1 and 2.

@Noc2  
These are the smart contracts:
Reputation Token Contract
Escrow Contract
Audit Directory

**Comment by keeganquigley** [comment]
Thanks for the application @ParthChaudhary31 a couple of questions:

- How do fewer smart contracts make it more decentralized? In my opinion, gas fees aren't typically an issue in our ecosystem, despite being more computationally expensive than substrate runtime weights. 
- What framework will you use to develop the front-end?
- What kind of reputation token contract will you use? Is it based on say the OpenBrush PSP37 standard NFT? Or is it more of a Soul Bound Token (SBT)?

Thanks!

**Comment by ParthChaudhary31** [comment]
> Thanks for the application @ParthChaudhary31 a couple of questions:
> 
> * How do fewer smart contracts make it more decentralized? In my opinion, gas fees aren't typically an issue in our ecosystem, despite being more computationally expensive than substrate runtime weights.
> * What framework will you use to develop the front-end?
> * What kind of reputation token contract will you use? Is it based on say the OpenBrush PSP37 standard NFT? Or is it more of a Soul Bound Token (SBT)?
> 
> Thanks!

@keeganquigley, To answer your questions:

1) There is no inverse relation between the number of smart contracts and decentralization for certain. We have limited the use of smart contracts to only where they're necessary to keep the process decentralized. This is to help the platform ease user onboarding, The fact that auditors will not have to make an initial investment (However low) to accept a bid will make the platform feel much more open and remove whatever little entry barrier.
  
2) We will use the ReactJs library for the development of our front end.

3) As for reputation tokens, We're planning to create them as SBTs (Soul BoundTokens).

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the update. Could you integrate React, etc. also, into the application? Once this is done, I will mark it as ready for review.

**Review by Noc2 — APPROVED** [review]
Thanks for the update. As promised, I will mark it as ready for review, and I'm personally willing to go ahead with it. However, it might help to reduce the price. Currently, you need five approvals, and I'm not sure everyone will be willing to accept the current rate.

**Review by keeganquigley — DISMISSED** [review]
Thanks @ParthChaudhary31, for your thorough answers and for reducing the price. I am happy to support it as well.

**Comment by ParthChaudhary31** [comment]
> Thanks for the update. As promised, I will mark it as ready for review, and I'm personally willing to go ahead with it. However, it might help to reduce the price. Currently, you need five approvals, and I'm not sure everyone will be willing to accept the current rate.

@Noc2  We'd really like to contribute to Polkadot's community as a firm and hence are willing to revise our price. As for the price per engineer, I have offered updated pricing, considering the current market, which would bring the grant amount down to $40,000 (~ $7k less than the last quote), if we remove all our margin we can further decrease the price to $36,500. Beyond this, we are working solely for the growth of the Polkadot community, Our final price after all consideration would be $29,920.

**Review by Noc2 — APPROVED** [review]
Thanks a lot for the update.

File: applications/Security_Marketplace.md [hunk 1]
```diff
@@ -0,0 +1,293 @@
+# Security Marketplace
+
+- **Team Name:** Antier Solutions
+- **Payment Address:** 1XhNxALoCmXF86aiwZtdDJspjWZBiRX9UfbJ4jQsWN15eSu (aUSD)
+- **[Level](https://github.com/w3f/Grants-Program/tree/master#level_slider-levels):** 3
```
Comment (takahser) [inline applications/Security_Marketplace.md:5]:
```suggestion
- **[Level](https://github.com/w3f/Grants-Program/tree/master#level_slider-levels):** 2
```

**Review by takahser — CHANGES_REQUESTED** [review]
@ParthChaudhary31 thanks for the update. In general, I'm happy to approve as well, but could you change the grant level to 2 (instead of 3) since the final price is now less than $30k and it'll be easier to get 3 approvals than 5 approvals.

**Review by takahser — CHANGES_REQUESTED** [review]
@ParthChaudhary31 thanks for the update. In general, I'm happy to approve as well, but could you change the grant level to 2 (instead of 3) since the final price is now less than $30k and it'll be easier to get 3 approvals than 5 approvals.

**Review by keeganquigley — APPROVED** [review]

**Review by takahser — APPROVED** [review]

**Comment by ParthChaudhary31** [comment]
Thank you very much @Noc2, @keeganquigley & @takahser for your vote of confidence, We are ecstatic to contribute to the community as a firm. We have aligned the team and kick-started the development process for milestone 1.

**Comment by Noc2** [comment]
@ParthChaudhary31, thanks for pinging us here. In this case, could you create a PR to update the original application?

**Comment by ParthChaudhary31** [comment]
> @ParthChaudhary31, thanks for pinging us here. In this case, could you create a PR to update the original application?

Sure thing, is there a standard process I need to follow for that? or should I create a new application PR for this with suggested amendments?

**Comment by semuelle** [comment]
> is there a standard process I need to follow for that? or should I create a new application PR for this with suggested amendments?

Yeah, it's like the application PR, but instead of a new document, you just commit changes to the existing one.

**Comment by ParthChaudhary31** [comment]
Raised a new PR here : 
https://github.com/w3f/Grants-Program/pull/1901

**Comment by keeganquigley** [comment]
Hi @ParthChaudhary31 is it possible to get an update on milestone 1?

**Comment by ParthChaudhary31** [comment]
> Hi @ParthChaudhary31 is it possible to get an update on milestone 1?

@keeganquigley We have submitted Milestone 1, you can check it out [Here](https://github.com/w3f/Grant-Milestone-Delivery/pull/945/files).

We're currently in the process of testing and optimizing Deliverables of milestone 2 and will raise a PR for the same by the starting week of next month.

**Comment by KennethkLopez** [comment]
Hey @ParthChaudhary31 & @semuelle are you guys in search of a Product Designer for the marketplace? Would love to chat on how I can contribute. Here's my [website](https://kennethk.io/) for reference. 

Looking forward to connecting!

**Pull request merged**